### PR TITLE
[book] fix the cycle fallback section to match current code

### PR DESCRIPTION
**Pull request by carljm**
Update the book section on cycle fallbacks to match what actually worked for me with the current Salsa code in main branch.

### CODE REVIEW

File: book/src/cycles/fallback.md [hunk 1]
```diff
@@ -2,17 +2,16 @@
...
+To use cycle recovery, you annotate potential participants in the cycle with the `recovery_fn` argument to `#[salsa::tracked]`, e.g. `#[salsa::tracked(recovery_fn=my_recovery_fn)]`. When a cycle occurs, if any participant P has recovery information, then no panic occurs. Instead, the execution of P is aborted and P will execute the recovery function to generate its result. Participants in the cycle that do not have recovery information continue executing as normal, using this recovery result.
 
-The recovery function has a similar signature to a query function. It is given a reference to your database along with a `salsa::Cycle` describing the cycle that occurred; it returns the result of the query. Example:
+The recovery function has a similar signature to a query function. It is given a reference to your database along with a `salsa::Cycle` describing the cycle that occurred and the input ingredient to the query that caused the cycle; it returns the result of the query. Example:
```
Comment (nikomatsakis) [inline book/src/cycles/fallback.md:7]:
```suggestion
The recovery function has a similar signature to a query function. It is given a reference to your database along with a `salsa::Cycle` describing the cycle that occurred and the arguments to the tracked function that caused the cycle; it returns the result of the query. Example:
```

File: book/src/cycles/fallback.md [hunk 2]
```diff
@@ -2,17 +2,16 @@
...
 
 ```rust
 fn my_recover_fn(
     db: &dyn MyDatabase,
     cycle: &salsa::Cycle,
+    input: MyIngredient,
```
Comment (nikomatsakis) [inline book/src/cycles/fallback.md:13]:
```suggestion
    arg1: T1,
    ...
    argN: TN,
```

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
Looks great, left a few nits. Is there a test case or example we can point people at?

**Comment by carljm** [comment]
> Is there a test case or example we can point people at?

Good call, I'll add a link to https://github.com/salsa-rs/salsa/blob/cd339fc1c9a6ea0ffb1d09bd3bffb5633f776ef3/tests/cycles.rs#L132-L141

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**